source/n_fields_problem.cc

```diff
@@ -457,16 +490,16 @@ void NFieldsProblem<dim, spacedim, n_components, LAC>::refine_mesh ()
                                           0,
                                           triangulation->locally_owned_subdomain());
 
-      parallel::distributed::GridRefinement::
-      refine_and_coarsen_fixed_fraction (*triangulation,
-                                         estimated_error_per_cell,
-                                         0.3, 0.1);
-
-
+      if ((typeid(typename LAC::VectorType) == typeid(pVEC)))
+        pgf.mark_cells(estimated_error_per_cell, *triangulation);
+      else
+        pgf.mark_cells(estimated_error_per_cell, *triangulation);
+      // parallel::distributed::GridRefinement::
+      // refine_and_coarsen_fixed_fraction (*triangulation,
+      //                                    estimated_error_per_cell,
+      //                                    0.3, 0.1);
```

Comment:
Parsed grid refinement does **not** do the refinement for you. You have to call refine_and_coarsen yourself...